sourcegraph/learn
# Component/atom/layout reorg

### What should this PR do?
Resolves [DEVED-21](https://sourcegraph.atlassian.net/browse/DEVED-21)  moving smaller components into an `atoms` folder, layouts into a `layouts` folder, and templates into a `templates` folder. 

The ticket has a linked doc that explains the logic in more detail, but the basic premise is that pages should render scoped, genre-specific templates, which then set the context for components. Components are then the "dumb" building blocks that begin to set styles, but which do not perform things like data fetching or include business logic. Components are composed of atoms, which in turn set repeatable styles that are usable across multiple components.

For now, we are not taking a hard stance of composability — as in, components do not necessarily need to be composed entirely of atoms etc.

### Why are we making this change?
- The goals are to ensure standardization across the app, which is more likely if we are reusing smaller pieces across components, and to avoid any duplication (of say, styles).

### What are the acceptance criteria? 
- The organization makes sense and seems reasonable.
- The app functions as it currently does in prod.

### How should this PR be tested?
- Check out the branch and check that the app functions properly.
- Offer any feedback on organization.

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.
